### PR TITLE
CLN: Remove unnecessary code

### PR DESCRIPTION
**Pull request by suminb**
`frozenset()` and `frozenset([])` essentially produce identical results, and thus there is no need to pass an empty list to create an empty `frozenset`.

- [ ] tests added / passed (CI is still running...)
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry (no new feature)

### CODE REVIEW

**Comment by jreback** [comment]
ok sure

File: pandas/tests/dtypes/test_inference.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ def coerce(request):
     ({'a', 1},                  'set', 'set'),                  # noqa: E241
     (set(),                     'set', 'set-empty'),            # noqa: E241
     (frozenset({'a', 1}),       'set', 'frozenset'),            # noqa: E241
-    (frozenset([]),             'set', 'frozenset-empty'),      # noqa: E241
+    (frozenset(),               'set', 'frozenset-empty'),      # noqa: E241
```
Comment (jbrockmendel) [inline pandas/tests/dtypes/test_inference.py]: Is the noqa still needed?

Comment (jreback) [reply pandas/tests/dtypes/test_inference.py]: yes, this has to do with liniing things up.

**Comment by jreback** [comment]
thanks @suminb

**Pull request merged**